configs/common/packages.yaml

```diff
@@ -54,17 +54,19 @@ packages:
   # Also, check the acorn and derecho site configs which have esmf modifications.
   esmf:
     require:
-      - '~xerces ~pnetcdf +shared +external-parallelio +python'
-      - any_of: ['@=8.6.1 snapshot=none', '@=8.7.0b11 snapshot=b11']
-      - any_of: ['fflags="-fp-model precise" cxxflags="-fp-model precise"']
-        when: "%intel"
-        message: "Extra ESMF compile options for Intel"
-      #- any_of: ['']
-      #  when: "%gcc"
-      #  message: "Extra ESMF compile options for GCC"
-      #- any_of: ['']
-      #  when: "%apple-clang"
-      #  message: "Extra ESMF compile options for GCC"
+    - '~xerces ~pnetcdf +shared +external-parallelio'
+    - any_of: ['@=8.6.1 snapshot=none', '@=8.7.0b11 snapshot=b11']
+    - any_of: ['fflags="-fp-model precise" cxxflags="-fp-model precise"']
+      when: "%intel"
+      message: "Extra ESMF compile options for Intel"
+    #- any_of: ['']
+    #  when: "%gcc"
+    #  message: "Extra ESMF compile options for GCC"
+    #- any_of: ['']
+    #  when: "%apple-clang"
+    #  message: "Extra ESMF compile options for GCC"
+    prefer:
+    - '+python'
```

Comment:
We want to be able to turn on/off the `python` variant, but the default should be to have it ??? enabled? disabled? (e.g. for ufs-weather-model-env or a basic neptune-env).

What do you think the default should be? Should the variant be enabled in the unified environment? There was a request for the ESMF Python interface in spack-stack in the last few months @AlexanderRichert-NOAA @RatkoVasic-NOAA @srherbener

Because we currently use different ESMF versions (8.6 vs 8.7) in UFS and NEPTUNE, the unified environment with the above change contains `esmf@8.6.1+python` and `esmf@8.7.0b11 +python`.

Reply:
Does it have to be a snapshot now that 8.7.0 was released?

Reply:
I can add the spack submodule ptr update to this PR so that we can bump it to just 8.7.0, yes. Next week!

Reply:
Done in https://github.com/JCSDA/spack-stack/pull/1318/commits/41b59e35d7358e2418f1bbad04a2208928ef7be5

Reply:
CI tests all use esmf@8.7.0 for NEPTUNE and esmf@8.6.1 for UFS